### PR TITLE
Default sync days setting

### PR DESCRIPTION
**Pull request by adnanhashmi09**
## Proposed changes (including videos or screenshots)
- Made `DEFAULT_SNC_DAYS_SETTINGS` part of the settings module.
- Now we wouldn't read the default sync days value from the env variable. Instead we save the default sync days in the database and get the value from there. 
- If value is not found in the database default value of `31` is used.
- Necessary model, dataclasses and adapters were modified and created. 
- The usages were changed to include `get_settings` function from the settings service

## Testing
- The code was testing by calling the following API to save the `DEFAULT_SYNC_DAYS_SETTING`:
```bash
curl --location --request PUT 'http://localhost:9696/orgs/{org_id}/settings' \
--header 'Content-Type: application/json' \
--data '{
    "setting_type": "DEFAULT_SYNC_DAYS_SETTING",
    "setting_data": {"default_sync_days": 100}
}'
```
- And then triggering the sync. The data was then verified to have been synced for past 100 days

### CODE REVIEW

File: backend/analytics_server/mhq/service/incidents/sync/etl_handler.py [hunk 1]
```diff
@@ -54,9 +59,19 @@ def sync_org_incident_services(self, org_id: str):
             LOG.error(f"Error syncing incident services for org {org_id}: {str(e)}")
             return
 
-    def _sync_service_incidents(self, service: OrgIncidentService):
+    def _sync_service_incidents(self, org_id: str, service: OrgIncidentService):
```
Comment (samad-yar-khan) [inline backend/analytics_server/mhq/service/incidents/sync/etl_handler.py:62]: Can't we use service.org_id

**Review by samad-yar-khan — CHANGES_REQUESTED** [review]

File: backend/analytics_server/mhq/service/settings/configuration_settings.py [hunk 2]
```diff
@@ -58,6 +59,13 @@ def _adapt_incident_types_setting_from_setting_data(
             ]
         )
 
+    def _adapt_default_sync_days_setting_setting_from_setting_data(
```
Comment (Kamlesh72) [inline backend/analytics_server/mhq/service/settings/configuration_settings.py:62]: "setting" word is repeated.
\_adapt_default_sync_days_<mark>setting_setting</mark>_from_setting_data

File: backend/analytics_server/mhq/service/incidents/sync/etl_handler.py (hunk 1 above)
Comment (adnanhashmi09) [reply backend/analytics_server/mhq/service/incidents/sync/etl_handler.py:62]: yeah good point. Thanks

File: backend/analytics_server/mhq/service/settings/configuration_settings.py (hunk 2 above)
Comment (adnanhashmi09) [reply backend/analytics_server/mhq/service/settings/configuration_settings.py:62]: Oh. good catch @Kamlesh72 
thanks mate

**Review by samad-yar-khan — APPROVED** [review]

**Pull request merged**